samples/javascript_nodejs/06.using-cards/dialogs/mainDialog.js

```diff
@@ -1,170 +1,169 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-const { AttachmentLayoutTypes, ActivityTypes, CardFactory } = require('botbuilder');
-const { ChoicePrompt, DialogSet, DialogTurnStatus, ListStyle } = require('botbuilder-dialogs');
+const { AttachmentLayoutTypes, CardFactory } = require('botbuilder');
+const { ChoicePrompt, ComponentDialog, DialogSet, DialogTurnStatus, ListStyle, WaterfallDialog, WaterfallStepContext } = require('botbuilder-dialogs');
+const AdaptiveCard = require('../resources/adaptiveCard.json');
 
-/**
- * RichCardsBot prompts a user to select a Rich Card and then returns the card
- * that matches the user's selection.
- */
-class RichCardsBot {
-    /**
-     * Constructs the three pieces necessary for this bot to operate:
-     * 1. StatePropertyAccessor
-     * 2. DialogSet
-     * 3. ChoicePrompt
-     *
-     * The only argument taken (and required!) by this constructor is a
-     * ConversationState instance.
-     * The ConversationState is used to create a BotStatePropertyAccessor
-     * which is needed to create a DialogSet that houses the ChoicePrompt.
-     * @param {ConversationState} conversationState The state that will contain the DialogState BotStatePropertyAccessor.
-     */
-    constructor(conversationState) {
-        // Store the conversationState to be able to save state changes.
-        this.conversationState = conversationState;
-        // Create a DialogState StatePropertyAccessor which is used to
-        // persist state using dialogs.
-        this.dialogState = conversationState.createProperty('dialogState');
+const MAIN_WATERFALL_DIALOG = 'mainWaterfallDialog';
+
+class MainDialog extends ComponentDialog {
+    constructor(logger) {
+        super('MainDialog');
+
+        if (!logger) {
+            logger = console;
+            logger.log('[MainDialog]: logger not passed in, defaulting to console');
+        }
 
-        // Create a DialogSet that contains the ChoicePrompt.
-        this.dialogs = new DialogSet(this.dialogState);
+        this.logger = logger;
 
-        // Create the ChoicePrompt with a unique id of 'cardPrompt' which is
-        // used to call the dialog in the bot's onTurn logic.
-        const prompt = new ChoicePrompt('cardPrompt');
+        // Define the main dialog and its related components.
+        this.addDialog(new ChoicePrompt('cardPrompt'))
+            .addDialog(new WaterfallDialog(MAIN_WATERFALL_DIALOG, [
+                this.choiceCardStep.bind(this),
+                this.showCardStep.bind(this)
+            ]));
```

Comment:
this chaining is a little silly if you ask me - you're inside the constructor

Reply:
Fixed